src/transformer.cpp

```diff
@@ -53,20 +57,9 @@ void Transformer::send(const std::string& id, TimeValue tv)
     data_channel_->publish(data_exchange_, id, DataChunk(tv).SerializeAsString());
 }
 
-void Transformer::config(const json& config)
+void Transformer::configure(const json& config)
```

Comment:
Who's responsible for the `unsubscribe` in the case of an reconfigure?

Reply:
I'd say the `Transformer` itself is, but it turns out that there does not exist a `transformer.unsubscribe` RPC. I guess we missed that in the manager?

In theory I'd unsubscribe from all input metrics (`Transformer::input_metrics_`), before calling `on_transformer_config`. We just have to make sure the corresponding data queue gets dropped in an orderly fashion. I'll have to wrap my head around how to do that.

Reply:
It's not easily solved by adding an `unsubscribe`. Since the transformer uses a persistent queue, the initial subscribe doesn't start with a clean slate. But the transformer cannot possibly know which metrics it would need to unsubscribe during restart.

So we would need a "`unsubscribe_all_existing_and_unused`" option for `transformer.subscribe` so that the manager can do that. Which is also difficult because there's not AMQP command to get queue bindings (either needs `rabbitmqctl` or manager-state).

If we had that option for `transformer.subscribe`, we might as well use it also for reconfigure instead of a new `transformer.unsubscribe`.

In summary: It's complicated, we need to deal with this eventually, but I consider this out of scope for now. In this PR you may assume it is not necessary - keep some `TODO`s.